multimap/src/main/java/org/infinispan/multimap/impl/SetBucket.java

```diff
@@ -154,4 +162,22 @@ public boolean equals(Object o) {
    public int hashCode() {
       return Objects.hash(values);
    }
+
+   public final class SetBucketResult<T> {
```

Comment:
Can just be a record.

Reply:
To make it a record I'll need to add another generic here for the `SetBucket<V>`. I didn't make these classes static to have just a single generic for the result.

Reply:
That should be fine, the class is static at that point. It should reduce the number of lines added.